src/components/src/main/java/org/apache/jmeter/control/IncludeController.java

```diff
@@ -92,6 +94,32 @@ public String getIncludePath() {
         return this.getPropertyAsString(INCLUDE_PATH);
     }
 
+    /**
+     * return the JMX file path with function support.
+     * @return the JMX file path with function support
+     */
+    public String getIncludePathAsFunction()
+    {
+        String jmxfile = this.getPropertyAsString(INCLUDE_PATH);
+        CompoundVariable masterFunction = new CompoundVariable();
+        try{
+            log.debug("Trying to evaluate 'Include Path' as an expression: {}", jmxfile);
+            masterFunction.setParameters(jmxfile);
+            if(masterFunction.hasFunction()) {
+                String jmxfileCompile = masterFunction.getFunction().execute();
+                log.debug("The value of 'Include Path' is computed as: {}", jmxfileCompile);
+                return jmxfileCompile;
+            }
+        } catch (InvalidVariableException e)
+        {
+            log.warn("Invalid variable in 'Include Path' {}. See log for details", jmxfile);
+            log.warn("Invalid variable in 'Include Path':", e);
+        }
+
+        log.debug("The value of 'Include Path' is simple string: {}", jmxfile);
+        return jmxfile;
+    }
```

Comment:
I think this is the wrong approach to attack the problem. It looks like `includePath.setRunningVersion(true)`.

Reply:
if apply `includePath.setRunningVersion(true)`
and set includePath like `${__P(somePropName, 'defaultValue')}`
and open JMX File in GUI Mode

then includePath view like value of `somePropName` or `'defaultValue'`, but not view as `${__P(somePropName, 'defaultValue')}`

Reply:
Function `getIncludePathAsFunction()` did not change value of `includePath`. Function `getIncludePathAsFunction()` used insdead of simple `this.getPropertyAsString(INCLUDE_PATH)` for valiable an functions support without change of value `includePath`

Reply:
AFAIK `setRunningVersions(tue)` should be called just before script execution (like it is called for other elements)

Reply:
Ok. I try tested this